core/src/main/java/org/apache/druid/data/input/impl/JsonInputFormat.java

```diff
@@ -73,7 +88,14 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
-    return new JsonReader(inputRowSchema, source, getFlattenSpec(), objectMapper, keepNullColumns);
+    return this.lineSplittable ?
+           new JsonLineReader(inputRowSchema, source, getFlattenSpec(), objectMapper, keepNullColumns) :
+           new JsonReader(inputRowSchema, source, getFlattenSpec(), objectMapper, keepNullColumns);
+  }
+
+  public void setLineSplittable(boolean lineSplittable)
```

Comment:
This doesn't look pretty, but simple enough to handle this exceptional case. Maybe we can introduce another layer on top of inputFormat to make it more pretty, but I don't think it will be worth at least at this point.